### PR TITLE
BACKEND| 095 validate screenings date

### PR DESCRIPTION
**Pull request by synowa**

### CODE REVIEW

File: api/screening/screeningRoute.js [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
+import { check, validationResult } from 'express-validator';
 import Screening from './Screening.js';
 import Movie from '../movie/Movie.js';
 import CinemaHall from '../cinemaHall/CinemaHall.js';
 import authMiddleware from '../authentication/authMiddleware.js';
 import adminMiddleware from '../admin/adminMiddleware.js';
 
 const router = express.Router();
+const modifyDate = (startDate, minutes) =>
+  new Date(startDate.getTime() + minutes * 60000);
```
Comment (vieraboschkova) [inline api/screening/screeningRoute.js]: Could we move this to utils?

File: api/screening/screeningRoute.js [hunk 2]
```diff
@@ -99,31 +142,30 @@ router
...
     } catch (e) {
-      res.status(400).send(e);
+      return res.status(400).send(e);
     }
   })
   // @route DELETE api/screenings/id
   // @description Delete a screening
-  // @access admin
+  // @access Admin
   .delete(authMiddleware, adminMiddleware, async (req, res) => {
```
Comment (vieraboschkova) [inline api/screening/screeningRoute.js:161]: Should probably remove also all the orders and tickets connected to this screening and also send email to clients that the screening was cancelled.

File: api/screening/screeningRoute.js [hunk 3]
```diff
@@ -22,69 +25,109 @@ router
...
+                $lte: modifyDate(startDate, movie.duration + 10),
+              },
+              cinemaHall: cinemaHallId,
+            },
+            {
+              endDate: {
+                $gte: modifyDate(startDate, -10),
+                $lte: modifyDate(startDate, movie.duration),
+              },
+              cinemaHall: cinemaHallId,
```
Comment (vieraboschkova) [inline api/screening/screeningRoute.js:80]: Great you're checking for conflicted screenings. A check we should add is a check with the cinemas opening hours probably.

**Review by vieraboschkova — CHANGES_REQUESTED** [review]
See if you can add additional checks and move the function to modify date to utils.

File: api/screening/screeningRoute.js (hunk 1 above)
Comment (synowa) [reply api/screening/screeningRoute.js]: No problem.

**Comment by synowa** [comment]
> See if you can add additional checks and move the function to modify date to utils.

Let me do that in a separate PR 🙂

File: api/screening/screeningRoute.js (hunk 2 above)
Comment (synowa) [reply api/screening/screeningRoute.js:161]: In our case it was only possible to remove connected tickets for now. Emails - done.

File: api/screening/screeningRoute.js (hunk 3 above)
Comment (synowa) [reply api/screening/screeningRoute.js:80]: I'll work on that in a separate PR.

**Review by paula0403 — APPROVED** [review]

**Review by vieraboschkova — APPROVED** [review]

**Pull request merged**